packages/oncokb-frontend-commons/src/components/OncoKB.tsx

```diff
@@ -114,73 +114,44 @@ function findDefaultDataTypeForTooltip(
     return OncoKbCardDataType.BIOLOGICAL;
 }
 
-@observer
-export default class OncoKB extends React.Component<IOncoKbProps, {}> {
-    constructor(props: IOncoKbProps) {
-        super(props);
-        makeObservable(this);
-    }
-
-    @observable showFeedback: boolean = false;
-    @observable tooltipDataLoadComplete: boolean = false;
-
-    public render() {
-        let oncoKbContent: JSX.Element;
+export const OncoKB: React.FunctionComponent<IOncoKbProps> = (
+    props: IOncoKbProps
+) => {
+    const [showFeedback, updateFeedback] = React.useState(false);
+    const [tooltipDataLoadComplete, updateTooltipDataLoad] = React.useState(
+        false
+    );
 
-        if (this.props.status === 'error') {
-            oncoKbContent = errorIcon('Error fetching OncoKB data');
-        } else if (this.props.status === 'pending') {
-            oncoKbContent = loaderIcon('pull-left');
-        } else {
-            oncoKbContent = this.props.mergeAnnotationIcons
-                ? this.singleAnnotationIcon()
-                : this.multiAnnotationIcon();
-
-            if (!this.props.disableFeedback && this.showFeedback) {
-                oncoKbContent = (
-                    <span>
-                        {oncoKbContent}
-                        <OncoKbFeedback
-                            userEmailAddress={this.props.userEmailAddress}
-                            hugoSymbol={this.props.hugoGeneSymbol}
-                            alteration={
-                                this.props.indicator
-                                    ? this.props.indicator.query.alteration
-                                    : undefined
-                            }
-                            showFeedback={this.showFeedback}
-                            handleFeedbackClose={this.handleFeedbackClose}
-                        />
-                    </span>
-                );
-            }
-        }
+    function handleFeedbackOpen(): void {
+        updateFeedback(showFeedback => (showFeedback = true));
+    }
 
-        return oncoKbContent;
+    function handleFeedbackClose(): void {
```

Comment:
we can clean up the body of this function by moving these little functions outside (i.e. above ln 117).  Also, apply React.useCallback hook to memoize them will give us an optimization similar to what we got with mobx computed.   This is probably negligible in this case but still a good practice to adopt in our functional components.